ios/RNFIRMessaging.m

```diff
@@ -247,12 +268,46 @@ - (void)messaging:(nonnull FIRMessaging *)messaging didRefreshRegistrationToken:
     [self sendEventWithName:FCMTokenRefreshed body:fcmToken];
 }
 
+- (void)handleRegisterUserNotificationSettings:(NSNotification *)notification
+{
+    if (_requestPermissionsResolveBlock == nil) {
+      return;
+    }
+
+    UIUserNotificationSettings *notificationSettings = notification.userInfo[@"notificationSettings"];
+
+    if (((notificationSettings.types & UIUserNotificationTypeAlert) == 0) ||
```

Comment:
I think there is a better way to do this in general, which is to pass what notifications you want to requestPermissions and then resolve/reject based on those.  I will try and cook that up now.